drivers/net/tap/rte_eth_tap.c

```diff
@@ -1182,6 +1185,13 @@ tap_dev_close(struct rte_eth_dev *dev)
 	struct pmd_internals *internals = dev->data->dev_private;
 	struct pmd_process_private *process_private = dev->process_private;
 	struct rx_queue *rxq;
+	int deviceNS = -1;
+	int originalNS = -1;
+
+	if (dev->data->netns[0] != '\0')
+	{
+		rte_ethdev_switch_netns(dev->data->netns, &deviceNS, &originalNS);
```

Comment:
what about the return value of rte_ethdev_switch_netns

Reply:
I don't understand why are you doing the switch on dev_close? Isn't exit_ns enough?